components/clarinet-deployments/src/lib.rs

```diff
@@ -472,8 +451,6 @@ pub async fn generate_default_deployment(
             let (epoch, clarity_version) =
                 get_boot_contract_epoch_and_clarity_version(contract_name.as_str());
 
-            // Set the session to the correct epoch for validation
-            session.update_epoch(epoch);
```

Comment:
@tippenein We commented a lot on this line during the override boot-contracts PR.

I still think that this is probably not necessary because we only call `interpreter.build_ast(&contract)`, which calls `clarity::vm::ast::build_ast_with_diagnostics`, which takes the epoch as an arguments (provided in `&contract`), and doesn't read if from the state of the session (or the datastore).

So I think we're fine. I still have to run more manuel tests on the PR, and especially makes sure that it properly handles boot contracts override 👍